src/foam/nanos/notification/SendNotificationDAO.java

```diff
@@ -51,24 +51,52 @@ public void put(Object o, Detachable d) {
         @Override
         public void put(Object o, Detachable d) {
           User user = (User) o;
-          send(user, notif, x);
+          send(x, user, notif);
         }
       });
     }
 
     if ( SafetyUtil.isEmpty(notif.getGroupId()) && ! notif.getBroadcasted() ) {
-      return super.put_(x, notif);
+      Notification notification = notif;
+      User nofifUser = (User) userDAO.find(notif.getUserId());
+      if ( nofifUser != null ) {
+        NotificationSetting notificationSetting = getNotificationSetting(x, nofifUser);
+
+        // We cannot permanently disable in-app notifications, so mark them read automatically
```

Comment:
curious why the InAppNotificationSettings can't handle setting 'read' during doNotify.

Reply:
sometimes we don't go through InAppNotificationSettings .. specifically when we save a user's notification directly to the down, hence the logic here as well as in the InAppNotification .. I can't really replace this with InAppNotificationSettings or I'll get an infinite loop.